File: app/src/main/java/com/ygorcesar/jamdroidfirechat/model/Message.java
package com.ygorcesar.jamdroidfirechat.model;

import java.util.HashMap;

public class Message {
    private String email;
    private String message;
    private int type;
    private HashMap<String, Object> time;
    private MapLocation mapLocation;

    public Message() {
    }

    /**
     * @param email
     * @param message
     * @param type 0 - TEXT; 1 - IMAGE; 2 - LOCATION
     * @param time
     */
    public Message(String email, String message, int type, HashMap<String, Object> time) {
        this.email = email;
        this.message = message;
        this.type = type;
        this.time = time;
    }

    public Message(String email, String message, int type, HashMap<String, Object> time, MapLocation mapLocation) {
        this.email = email;
        this.message = message;
        this.type = type;
        this.time = time;
        this.mapLocation = mapLocation;
    }

    public String getEmail() {
        return email;
    }

    public String getMessage() {
        return message;
    }

    public int getType() {
        return type;
    }

    public HashMap<String, Object> getTime() {
        return time;
    }

    public MapLocation getMapLocation() {
        return mapLocation;
    }
}

File: app/src/main/java/com/ygorcesar/jamdroidfirechat/data/entity/Message.kt
package com.ygorcesar.jamdroidfirechat.data.entity

import com.google.firebase.database.ServerValue
import com.ygorcesar.jamdroidfirechat.utils.ConstantsFirebase

data class Message(var email: String = "", var friendEmail: String = "", var message: String = "", var imgUrl: String = "", var type: Int = ConstantsFirebase.MessageType.TEXT,
                   var time: HashMap<String, Any> = hashMapOf(ConstantsFirebase.FIREBASE_PROPERTY_MESSAGE_TIME_SENDED to ServerValue.TIMESTAMP),
                   var mapLocation: MapLocation? = null)